### PR TITLE
feat(edc):[TRI-1064] Extract EDC components

### PR DESCRIPTION
**Pull request by ds-jkreutzfeld**
EDC client code is now contained in a separate maven module, so it can be reused in the ESS implementation. It was necessary to also extract some shared code into a common module to avoid circular dependencies.

### CODE REVIEW

**Review by ds-jhartmann — CHANGES_REQUESTED** [review]
See individual comments.
In general: the following tests are failing with exception
 `java.util.NoSuchElementException: Unsupported AspectType: ASSEMBLY_PART_RELATIONSHIP. [...]`
- RelationshipDelegateTest
- IrsFunctionalTest
- IrsItemGraphQueryServiceSpringBootTest

OpenApi comparison fails as well

File: edc-client/src/test/java/org/eclipse/tractusx/edc/CxTestDataAnalyzerTest.java [hunk 1]
```diff
@@ -49,11 +51,11 @@ class CxTestDataAnalyzerTest extends LocalTestDataConfigurationAware {
     private final CxTestDataContainer cxTestDataContainer;
 
     CxTestDataAnalyzerTest() throws IOException {
-        super();
-
         cxTestDataContainer = localTestDataConfiguration.cxTestDataContainer();
     }
 
+
+
```
[the remark this answers is not in the document]
Comment (ds-jkreutzfeld) [reply edc-client/src/test/java/org/eclipse/tractusx/edc/CxTestDataAnalyzerTest.java]: Resolved

File: edc-client/src/test/java/org/eclipse/tractusx/edc/CxTestDataAnalyzerTest.java [hunk 2]
```diff
@@ -19,10 +19,11 @@
  *
  * SPDX-License-Identifier: Apache-2.0
  ********************************************************************************/
-package org.eclipse.tractusx.irs.edc;
+package org.eclipse.tractusx.edc;
 
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.eclipse.tractusx.irs.configuration.local.CxTestDataContainer.CxTestData.*;
+import static org.mockito.Mockito.mock;
```
[the remark this answers is not in the document]
Comment (ds-jkreutzfeld) [reply edc-client/src/test/java/org/eclipse/tractusx/edc/CxTestDataAnalyzerTest.java]: Resolved

**Comment by ds-jkreutzfeld** [comment]
> See individual comments. In general: the following tests are failing with exception `java.util.NoSuchElementException: Unsupported AspectType: ASSEMBLY_PART_RELATIONSHIP. [...]`
> 
> * RelationshipDelegateTest
> * IrsFunctionalTest
> * IrsItemGraphQueryServiceSpringBootTest
> 
> OpenApi comparison fails as well

The tests are fixed now, except for the OpenAPI comparison. This is due to the incomplete state of ess-dev, which was merged into this branch. As soon as that is fixed with https://github.com/catenax-ng/tx-item-relationship-service/pull/125, we can update this branch and the test will hopefully be green.

File: edc-client/pom.xml [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.eclipse.tractusx.irs</groupId>
+            <artifactId>irs-common</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.yaml</groupId>
+            <artifactId>snakeyaml</artifactId>
+            <version>1.31</version>
```
[the remark this answers is not in the document]
Comment (ds-jkreutzfeld) [reply edc-client/pom.xml]: Done :)

**Review by ds-jhartmann — APPROVED** [review]

**Pull request merged**